packages/flutter/lib/src/services/platform_channel.dart

```diff
@@ -32,8 +32,31 @@ export 'message_codec.dart' show MessageCodec, MethodCall, MethodCodec;
 /// The statistics include the total bytes transmitted and the average number of
 /// bytes per invocation in the last quantum. "Up" means in the direction of
 /// Flutter to the host platform, "down" is the host platform to flutter.
+bool get profilePlatformChannels => kProfilePlatformChannels || (!kReleaseMode && debugProfilePlatformChannels);
```

Comment:
nit: The action on this function is not to "profile" .  `shouldProfilePlatformChannels` would be more appropriate.